engine/src/gfx/mesh_gfx.cpp

```diff
@@ -438,9 +438,9 @@ Mesh::~Mesh() {
         vector<Mesh *> *hashers = bfxmHashTable.Get(hash_name);
         vector<Mesh *>::iterator finder;
         if (hashers) {
-            for (size_t i = hashers->size() - 1; i >= 0; --i) {
-                if (hashers->at(i) == this) {
-                    hashers->erase(hashers->begin() + i);
+            for (auto hashItem = hashers->begin(); hashItem != hashers->end(); ++hashItem) {
+                if (*hashItem == this) {
+                    hashers->erase(hashItem);
```

Comment:
did this after the old loop somehow ended up with a really high number even though there was supposedly only a small couple of entries in there. Nice thing is that ` erase` can take an iterator so we can easily do this. 

This should be relatively safe.

Reply:
There was a reason why I had it counting down from the end of the `hashers` collection instead of starting from the beginning. I still don't think I had it completely right though. I don't think this code is completely right either.

There's probably no way to get this to work right at all until we get Vega Strike to stop randomly corrupting its own internal memory state. Which it hasn't, in any of the existing branches of code, as far as I can tell.

I suppose your version will do for now, @BenjamenMeyer .

Reply:
@stephengtuggy yeah... this solved *my* issue; but I agree - I'm not sure it's quite right... perhaps b/c the iterator isn't modified before the erase so that it can safely move to the *next* item in the loop now that I think about it. I'll update that.

Reply:
https://gist.github.com/BenjamenMeyer/f64f22b0acbc958fb267badde84b2c8c - safe way to do it

Reply:
fixed

Reply:
Ultimately, we're not living in 2001. We should probably stop caching so aggressively and just duplicate stuff if it makes the code simpler and more elegant. Just my two cents.

Reply:
@royfalk agreed. We can fix that up as we refactor going forward.